src/main/java/duke/command/MarkCommand.java

```diff
@@ -0,0 +1,41 @@
+package duke.command;
+
+import duke.exception.DukeIllegalArgumentException;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.util.Printable;
+
+/**
+ * Represents a handler for the mark and unmark commands.
+ * Encapsulates all set completion logic.
+ */
+public class MarkCommand extends Command {
+    private final boolean isDone;
+
+    /**
+     * Creates a handler for the mark or unmark command, depending on the supplied argument.
+     * @param args Arguments supplied to the command handler.
+     * @param isDone The new completion state that the handler should set a task to.
+     */
+    MarkCommand(String args, boolean isDone) {
+        super(args);
+        this.isDone = isDone;
+    }
+
+    @Override
+    public boolean execute(Printable linePrinter, TaskList taskList) throws DukeIllegalArgumentException {
```

Comment:
This is a overridden abstract method inherited from the `Command` abstract class. Since the module's coding standard does not explicitly say that it is required, I am following Google's Style Guide. [Section 7.3.2](https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required) grants an exception for Javadoc to not be present on overridden methods since Javadocs are also implicitly inherited from the superclass.